### PR TITLE
compare probe/src directly issues/49855

### PR DESCRIPTION
**Pull request by jubrad**
### What does this PR do?
This changes the tokenized approach for string comparison
and shifts it to direct comparison of the content and the line in question
to determine if any changes should be made during file.line insert/ensure before or after

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/49855

### Previous Behavior
Weird comparison of words that begin a line to determine if changes should be made

### New Behavior
Direct comparison on lines and content to be added.

### Tests written?
Manually tested by replacing site-packages files.
Adjusted unit tests to match new functionality.

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/file.py
Comment (Ch3LL) [inline salt/modules/file.py]: Just one question. Can you elaborate why you are not adding `rstrip('\n\r')` to probe as well?

I also added some tests for this PR here https://github.com/jubrad/salt/pull/1  if you can give that a review and merge so we can ensure this does not regress.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/modules/file.py
Comment (jubrad) [reply salt/modules/file.py]: Thanks, sorry I believe that we've implemented a workaround so there is less pressure on a fix here from us, as such I've completely stopped thinking about it. 

Thanks for adding the tests.

The probe is coming from an argument to the method. I assumed any newlines in it would be intentional thus shouldn't be stripped, although it would probably cause less confusion and better code to just strip both.

**Comment by Ch3LL** [comment]
thanks for those changes. looks like the windows tests are failing because it still references `_starts_till` in salt/modules/win_file.py. Can you update those references to `_line_match`?

**Comment by Ch3LL** [comment]
bump @jubrad

**Comment by Ch3LL** [comment]
also waiting on review from @waynew

**Comment by waynew** [comment]
Oh, yeah, this one!

I'm pretty sure this does what it says it does, but I also discovered uh... several other issues with this module that were introduced. (Assuming tests pass) I'm OK with merging this, because it *is* an improvement, but we also certainly need to fix some of the other historical artifacts here.

@jubrad are you interested in fixing some of the other issues surrounding this bug, or just trying to get this fix in as-is?

**Comment by waynew** [comment]
@jubrad Do you have time to work on this in the next day or two? We'd like to get this fix incorporated in time for our 2019.2.1 release. If not, I can go ahead and build off your work here. If I don't hear from you, I'll assume you're busy 😂

**Comment by dwoz** [comment]
re-run full all

**Comment by Ch3LL** [comment]
ping @waynew looks like there are some failures and we are trying to get this into 2019.2.1

**Comment by waynew** [comment]
re-run py2-centos-7

**Comment by waynew** [comment]
re-run full all

**Comment by jubrad** [comment]
@waynew sorry I was on vacation during your message then I just completely dropped the ball. Looks like you've made some significant changes here. I'm happy to help move this along as needed and I'll make sure to increase the volume of my alerts for this pr.

**Comment by waynew** [comment]
@jubrad now that we've switched to the master branch approach, https://github.com/saltstack/salt/pull/53888 has been rebased for master, so I'm going to go ahead and close this PR - feel free to watch that other PR, or reply on here if you have any questions!

**Pull request closed without merging**
